### PR TITLE
[16.0][ADD] edi_account_edifact_oca

### PR DESCRIPTION
**Pull request by QuocDuong1306**
Draft state

Depend on:
- #8

### CODE REVIEW

File: edi_account_edifact_oca/components/listener.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2023
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo.addons.component.core import Component
+
+
+class MyEventListener(Component):
```
Comment (nilshamerlinck) [inline edi_account_edifact_oca/components/listener.py]: nitpick: can probably find a better name

File: edi_account_edifact_oca/components/validate.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo.addons.component.core import Component
+
+
+class EDIStorageValidateComponent(Component):
+    _name = "edi.output.account.move.validate"
+    _inherit = "edi.component.validate.mixin"
+    _usage = "output.validate.account.move"
+
+    def validate(self, value):
```
Comment (nilshamerlinck) [inline edi_account_edifact_oca/components/validate.py]: validation makes sense for xml files to validate using a xsd, but for edifact I don't think we need it

File: edi_account_edifact_oca/demo/edi_exchange_type.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    usage: output.generate.account.move
+  validate:
+    usage: output.validate.account.move
+  send:
+    usage: storage.send
+  env_ctx:
+    default_file_ext: 'edi'
+    defaut_release: 'd96a'
+    default_doc_type: 'rfq'
+    default_confirm_order: false
```
Comment (nilshamerlinck) [inline edi_account_edifact_oca/demo/edi_exchange_type.xml]: ?

Comment (nilshamerlinck) [reply edi_account_edifact_oca/demo/edi_exchange_type.xml]: unrelated

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]

**Comment by nilshamerlinck** [comment]
where do you check that the invoice's customer is linked with an edi backend?

File: edi_account_edifact_oca/demo/edi_exchange_type.xml (hunk 3 above)
Comment (QuocDuong1306) [reply edi_account_edifact_oca/demo/edi_exchange_type.xml:22]: Ah, i forgot removing when copy/paste

**Comment by QuocDuong1306** [comment]
I updated (also title changed)

File: edi_account_edifact_oca/components/listener.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+_logger = logging.getLogger(__name__)
+
+
+class AccountMoveEdifactListener(Component):
+    _name = "account.move.event.listener.edifact"
+    _inherit = "base.event.listener"
+    _apply_on = ["account.move"]
+
+    def on_post_account_move(self, move):
+        """Add your code here about creation of record"""
```
Comment (nilshamerlinck) [inline edi_account_edifact_oca/components/listener.py]: old docstring?

**Review by nilshamerlinck — APPROVED** [review]

**Pull request closed without merging**